agent/test-agent-cli/src/save_results.rs

```diff
@@ -0,0 +1,59 @@
+use crate::error::{ArchiveSnafu, Result};
+use argh::FromArgs;
+use copy_dir::copy_dir;
+use log::error;
+use model::constants::TESTSYS_RESULTS_DIRECTORY;
+use snafu::ResultExt;
+use std::path::Path;
+use test_agent::DefaultClient;
+
+#[derive(Debug, FromArgs, PartialEq)]
+#[argh(
+    subcommand,
+    name = "save-results",
+    description = "save the results file or directory in tar archive"
+)]
+pub(crate) struct SaveResults {
+    #[argh(short = 'd', option, description = "directory to add in tar archive")]
+    directory: Vec<String>,
+    #[argh(short = 'f', option, description = "file to add in tar archive")]
+    file: Vec<String>,
+}
+
+impl SaveResults {
+    pub(crate) async fn run(&self, _k8s_client: DefaultClient) -> Result<()> {
+        for dir in &self.directory {
+            copy_dir(dir, TESTSYS_RESULTS_DIRECTORY.to_owned() + "/" + dir)
+                .context(ArchiveSnafu)?;
+        }
+
+        for file in &self.file {
+            self.move_file(file, TESTSYS_RESULTS_DIRECTORY.to_owned() + "/" + file)
+                .await?;
+        }
+
+        if self.directory.is_empty() && self.file.is_empty() {
+            error!("Invalid arguments were provided. One of `--file`, `--directory` must be used.");
+        }
+
+        Ok(())
+    }
+
+    async fn move_file<P, Q>(&self, from: P, to: Q) -> Result<()>
```

Comment:
Where is `self` used? If it isn't it can be moved out of the impl.

Reply:
To call that method as self.move_file, we have to pass self as a param else I will have to call it as save_results::SaveResults::move_file. Is there any other better way to solve this ?

Reply:
Since `move_file` has no real association with `SaveResults` other than that that's where we are using it, maybe it should just be moved out of the `SaveResults` implementation and just be its own standalone function that we can call.

Reply:
You should be able to move it out and call `move_file`. It shouldn't need the extra stuff.